pkg/scheduler/framework/plugins/tainttoleration/taint_toleration.go

```diff
@@ -146,16 +146,16 @@ func countIntolerableTaintsPreferNoSchedule(taints []v1.Taint, tolerations []v1.
 
 // Score invoked at the Score extension point.
 func (pl *TaintToleration) Score(ctx context.Context, state *framework.CycleState, pod *v1.Pod, nodeName string) (int64, *framework.Status) {
-	nodeInfo, err := pl.handle.SnapshotSharedLister().NodeInfos().Get(nodeName)
+	s, err := getPreScoreState(state)
 	if err != nil {
-		return 0, framework.AsStatus(fmt.Errorf("getting node %q from Snapshot: %w", nodeName, err))
+		return 0, framework.AsStatus(err)
 	}
-	node := nodeInfo.Node()
 
-	s, err := getPreScoreState(state)
+	nodeInfo, err := pl.handle.SnapshotSharedLister().NodeInfos().Get(nodeName)
```

Comment:
Another concern here is why do you think calling `getPreScoreState` first will be more efficient, they're all memory operations.

Reply:
I do this for two reasons:
1. Get node is not easy to make mistakes, put the ones that are easy to make mistakes in the front, and put the ones that are not easy to report errors in the back, so as to improve the efficiency
2. Node is closer to nodeinfo, making it easier for people who read the code to understand the code

Reply:
We used to store state in `Map`, so we'll lock/unlock the map when getting/setting values. I think this is the main reason why we get node first. Currently, we replace `Map` with `sync.Map` which is more efficient in read-more-write-less scenarios, but we may still lock the map when reading in some circumstances.

So in general,  I'm not convinced this is more efficient to get state first. But yes, variables should sit near to the calling codes. Glad to hear others' advices. 🤔

Reply:
Thanks for your suggestion. what you said makes sense. What do you think of this PR? If you think it's dispensable, I'll close it

Reply:
Prefer to close it then. ^ ^